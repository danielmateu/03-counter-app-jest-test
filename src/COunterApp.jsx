import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const COunterApp = ({ value }) => {

    console.log('render...');

    const [counter, setCounter] = useState(value)

    //handleAdd
    const hanldeAdd = () => { setCounter(counter + 1) }

    const handleMinus = () => { setCounter(counter - 1) }

    const handleReset = () => { setCounter(value) }

    return (
        <>
            <h1>CounterApp</h1>
            <h2> {counter} </h2>
            <div className="buttons">
                <button onClick={hanldeAdd}>+1</button>
                <button onClick={handleMinus}>-1</button>
                <button onClick={handleReset}>Reset</button>
            </div>
        </>
    )
}

//PropTypes
COunterApp.propTypes = {
    value: PropTypes.number.isRequired
}


