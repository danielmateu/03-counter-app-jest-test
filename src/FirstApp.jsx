import React from 'react'
import PropTypes from 'prop-types'


const newMessage = {
    title: '',
    subtitle: 1,
    body: 'React is working!',
    name: '',
    age: 30,
    hero: true,
    power: 'Super Speed',
};

// const getResult = () => {
//     return 1 + 1;
// }

export const FirstApp = ({title,subtitle, name}) => {

    


    return (
        <>
            <h1>{title}</h1>
            {/* <code>{JSON.stringify(newMessage)}</code> */}
            <p>{subtitle}</p>
            <p>{name}</p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number,
    
}

FirstApp.defaultProps = {
    title: 'Default Title',
    subtitle: 'Default Subtitle',
    name: 'Daniel Mateu'
}
