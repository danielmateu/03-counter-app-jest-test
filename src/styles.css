html,body{
    margin: 0;
    padding: 0;
    height: 100%;
    width: 100%;
    background-color: #333;
    font-family: 'Roboto', sans-serif;
    font-size: 16px;
    color: #f5f5f5;
    text-align: center;
}

button{
    background-color: #f5f5f5;
    color: #333;
    border: none;
    border-radius: 5px;
    padding: 10px 20px;
    font-size: 16px;
    font-weight: 500;
    cursor: pointer;

}

button:hover{
    background-color: #727272;
    color: #f5f5f5;
}

.buttons{
    display: flex;
    justify-content: center;
    gap: 2em;
}

