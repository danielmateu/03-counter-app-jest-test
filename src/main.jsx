import React from 'react'
import ReactDOM from 'react-dom/client'
import { COunterApp } from './COunterApp'
import { FirstApp } from './FirstApp'
import './styles.css'


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <FirstApp 
      title = 'Eres Gay?'
      // subtitle = {123456}
    /> */}

    <COunterApp
      value = {10}
    />
  </React.StrictMode>
)
