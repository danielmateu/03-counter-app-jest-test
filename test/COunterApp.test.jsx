import { render, screen } from "@testing-library/react"
import { COunterApp } from "../src/COunterApp";


describe('Pruebas en <CounterApp/>', () => {
    //Debe hacer match con el snapshot
    test('Debe hacer match con el snapshot', () => {

        const initialValue = 10
        const { container } = render(<COunterApp value={initialValue} />)
        expect(container).toMatchSnapshot()

    });

    //Debe mostrar el valor por defecto de 100
    test('Debe mostrar el valor por defecto de 100', () => {

        const { container } = render(<COunterApp value={100} />)
        expect(container).toMatchSnapshot()
        expect(screen.getByText(100)).toBeTruthy()

    })
})



