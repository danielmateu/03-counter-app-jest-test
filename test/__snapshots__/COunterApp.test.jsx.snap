// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    10
     
  </h2>
  <div
    class="buttons"
  >
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button>
      Reset
    </button>
  </div>
</div>
`;

exports[`Pruebas en <CounterApp/> Debe mostrar el valor por defecto de 100 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <div
    class="buttons"
  >
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button>
      Reset
    </button>
  </div>
</div>
`;
